src/components/ImportOnyxState/index.tsx

```diff
@@ -20,26 +21,31 @@ export default function ImportOnyxState({setIsLoading}: ImportOnyxStateProps) {
             return;
         }
 
+        rollbackOngoingRequest();
         setIsLoading(true);
+
         const blob = new Blob([file as BlobPart]);
         const response = new Response(blob);
 
         response
             .text()
             .then((text) => {
                 const fileContent = text;
+
                 const transformedState = cleanAndTransformState<OnyxValues>(fileContent);
+
                 const currentUserSessionCopy = {...session};
                 setPreservedUserSession(currentUserSessionCopy);
                 setShouldForceOffline(true);
-                Onyx.clear(KEYS_TO_PRESERVE).then(() => {
-                    Onyx.multiSet(transformedState).then(() => {
-                        setIsUsingImportedState(true);
-                        Navigation.navigate(ROUTES.HOME);
-                    });
-                });
+
+                return processStateImport(transformedState);
+            })
+            .then(() => {
+                setIsUsingImportedState(true);
+                Navigation.navigate(ROUTES.HOME);
             })
-            .catch(() => {
+            .catch((error) => {
+                console.error('Error importing state:', error);
                 setIsErrorModalVisible(true);
                 setIsLoading(false);
             });
```

Comment:
```suggestion
            .catch((error) => {
                console.error('Error importing state:', error);
                setIsErrorModalVisible(true);
            })
            .finally(() => {
                setIsLoading(false);
            });
```

to align between platform files